From 5d9fe791907d92b75e3d2b4213376507cfd1d965 Mon Sep 17 00:00:00 2001
From: Jeroen Hofstee <jhofstee@victronenergy.com>
Date: Sun, 25 Apr 2021 17:42:40 +0200
Subject: [PATCH 15/17] QSortFilterProxyModel does not sort in some cases

If the model is empty and new rows are inserted thereafter, they were not
sorted correctly. This fixes that.

see https://bugreports.qt.io/browse/QTBUG-35180
---
 src/gui/itemviews/qsortfilterproxymodel.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/gui/itemviews/qsortfilterproxymodel.cpp b/src/gui/itemviews/qsortfilterproxymodel.cpp
index da444a1b33..acc7b68063 100644
--- a/src/gui/itemviews/qsortfilterproxymodel.cpp
+++ b/src/gui/itemviews/qsortfilterproxymodel.cpp
@@ -1216,6 +1216,8 @@ void QSortFilterProxyModelPrivate::_q_sourceDataChanged(const QModelIndex &sourc
         sort_source_rows(source_rows_insert, source_parent);
         insert_source_items(m->proxy_rows, m->source_rows,
                             source_rows_insert, source_parent, Qt::Vertical);
+        if (dynamic_sortfilter && source_sort_column < 0 && proxy_sort_column >= 0 && update_source_sort_column())
+            sort();
     }
 }
 
-- 
2.25.1

